Source/tmsg.cpp

```diff
@@ -29,15 +29,15 @@ size_t tmsg_get(byte *pbMsg)
 	size_t len = head->hdr.bLen;
 	// BUGFIX: ignores dwMaxLen
 	memcpy(pbMsg, head->body, len);
-	std::free(head);
+	delete head;
 	return len;
 }
 
 void tmsg_add(byte *pbMsg, uint8_t bLen)
 {
 	TMsg **tail;
 
-	TMsg *msg = static_cast<TMsg *>(std::malloc(bLen + sizeof(*msg)));
+	TMsg *msg = reinterpret_cast<TMsg *>(new byte[bLen + sizeof(*msg)]);

```

Comment:
This is invalid: `new byte[]` is not guaranteed to be aligned on `alignof(TMsg)`. Use `new TMsg[]` instead.

Reply:
Oh I see what this is doing.. It is unfortunately a bit tricky to do corretly in C++.
You have to use both `new` and `delete` with the alignment argument: 

Something like:

```c++
new (std::align_val_t(alignof(TMsg))) byte[bLen + sizeof(TMsg)]
```

and the corresponding `::operator delete[]` with the same alignment value when destroying

This is because `malloc` is guaranteed to align to the largest primitive alignment but `new` is not

Reply:
...and also cast back to byte[] before deleting, to be kosher and avoid any potential UB...

This is more trouble than it's worth. I'm just going to throw bad_alloc if the result is nullptr.

Reply:
>  I'm just going to throw bad_alloc if the result is nullptr.

It won't be a nullptr, as it doesn't know that you're going to use it as `TMsg` from the `new byte[]` call alone. If the allocator returns a pointer with an insufficient alignment for `TMsg`, it can crash on access on some CPUs, such as: MIPS, ARM before v7, x86 if the code is auto-vectorized, etc..

Reply:
That's not what I meant. I kept malloc and checked the result.